starter-code/src/functions-and-arrays.js

```diff
@@ -11,14 +17,32 @@ var words = [
   'crackpot'
 ];
 
+function findLongestWord(array) {
+  
+  if (array == undefined) return 0;
+  
+  array.sort((a,b)=> b.length - a.length);
+  return array[0];
+  
+}
+
 // Calculating a Sum
 
 var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray(array) {
+  return array.reduce((a,b)=>a+b,0);
+}
+
 // Calculate the Average
 
 var numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(array) {
+  let len = array.length;
+  return len === 0 || array.includes(NaN) ? undefined : array.reduce((a,b)=>a+b,0)/len;
```

Comment:
Have you done this before? Instead of repeating the code, what about reusing the function?
```
return len === 0 || array.includes(NaN) ? undefined : sumArray(array)/len;
```

Reply:
I will in the future. I just thought I had to rewrite it.